platform/features/layout/src/LayoutController.js

```diff
@@ -99,9 +105,19 @@ define(
                 e.preventDefault();
             }
 
+            // Deselects the object that is no longer in the composition.
+            function deselectRemovedObject(composition, previousComposition) {
+                if (composition !== previousComposition) {
+                    var removedId = _.difference(previousComposition, composition);
+                    if (removedId.length === 1 && removedId[0] === self.selectedId) {
+                        self.clearSelection();
+                    }
+                }
```

Comment:
I think this could be simplified to:

```
if (composition.indexOf(self.selectedId) === -1) {
    self.clearSelection();
}
```

...which is also probably short enough to be moved in-line below. Looking at the previous composition doesn't seem necessary; if the selected ID isn't in the composition, the selection doesn't make sense.